qiita_core/configuration_manager.py

```diff
@@ -178,6 +178,7 @@ def _get_main(self, config):
         if not self.certificate_file:
             self.certificate_file = join(install_dir, 'qiita_core',
                                          'support_files', 'server.crt')
+        self.cookie_secret = config.get('main', 'COOKIE_SECRET')
```

Comment:
IMO, we should check if `self.cookie_secret` is `None` (as it is done for `self.key_file` below). If it is None, I would go back to the current behavior, generate a random one, but also raising a Warning to the user to let him now that it was not set, in case that he actually needs to set it (e.g. load balancing).

I don't think it is critical to the point of raising an error, since the usage of that config value is specific to some functionality (we didn't realize that we needed until we wanted to do load balancing), but not signing at all I think it's bad.